mmcls/models/classifiers/timm.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) OpenMMLab. All right reserved.
+import re
+from collections import OrderedDict
+from typing import List, Optional
+
+import torch
+import torch.nn.functional as F
+
+from mmcls.registry import MODELS
+from mmcls.structures import ClsDataSample
+from .base import BaseClassifier
+
+
+@MODELS.register_module()
+class TimmClassifier(BaseClassifier):
+
+    def __init__(self,
+                 *args,
+                 loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
+                 train_cfg: Optional[dict] = None,
+                 data_preprocessor: Optional[dict] = None,
+                 init_cfg: Optional[dict] = None,
+                 **kwargs):
+        if data_preprocessor is None:
+            data_preprocessor = {}
+        # The build process is in MMEngine, so we need to add scope here.
+        data_preprocessor.setdefault('type', 'mmcls.ClsDataPreprocessor')
+
+        if train_cfg is not None and 'augments' in train_cfg:
+            # Set batch augmentations by `train_cfg`
+            data_preprocessor['batch_augments'] = train_cfg
+
+        super().__init__(
+            init_cfg=init_cfg, data_preprocessor=data_preprocessor)
+        from timm.models import create_model
+        self.model = create_model(*args, **kwargs)
+        self.loss_module = MODELS.build(loss)
+
```

Comment:
```
If with_cp:
    self.model.set_grad_checkpointing()
```

timm also supports gradient checkpointing in latest versions. You may want to consider this.

Reply:
Updated